vulcan/lib/server/controllers/sessions_controller.rb

```diff
@@ -11,6 +11,65 @@ def session
   end
 
   def submit
+    if can_hijack?
```

Comment:
This probably needs a bit of scrutiny. As noted in the comments, apparently Firefox has very strict default timeouts for active connections (without streaming, a long workflow ran fine with curl, Postman, and Chrome for me, so it is a browser-specific issue). To work around Firefox automatically aborting the POST submit request, we do two things:

1. Immediately return headers stating a 200 JSON response.
2. Stream a null byte back every 5 seconds. This keeps the connection alive and Firefox happy, and also does not impact JSON-parsing the response on the client-side.

Reply:
Sounds like https://en.m.wikipedia.org/wiki/Comet_(programming), should be plenty of libs to support this.

Reply:
When we move to an async model, that should also remove the problem...in addition to just a general "comet"-type philosophy, I think another of our issues is that it can take more than 10 seconds to send back data since running the cells can take awhile. Since we're not sending individual cell results back as they finish, it looks like a hung connection. We still need to stream "filler, keep-alive data" back, whether the protocol is SSE or WebSocket or anything similar.

There are Rails / ActiveSupport libraries that support streaming SSE back like this, this implementation is just a bit closer to the metal and bare Rack interface since client-side we're expecting simple JSON at the moment.